File: 11_Container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int volume;
        int max = 0;
        for (int i = 0; i < height.size() - 1; i++) {
            for (int j = i + 1; j < height.size(); j++) {
                volume = (j - i) * std::min(height[i], height[j]);
                if (volume > max) {
                    max = volume;
                }
            }
        }
        return max;
    }
};

File: python/11_Container_with_most_water.py
class Solution(object):
    def maxArea(self, height):
        maximum, left, right = 0, 0, len(height) - 1
        while (left < right):
            maximum = max(maximum, ((right - left)*min(height[left],height[right])))
            if (height[left] <= height[right]):
                left += 1
            else:
                right -= 1
        return maximum